### DISCUSSION TITLE
lotus-miner  Unexpected termination，ask for help。

### DISCUSSION THREAD

**Discussion by ChaiWenyu1983** [filecoin-project/lotus#8839] [General Q&A] [answered]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [ ] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
lotus version 1.15.2
```

### Describe the Bug

[miner.log](https://github.com/filecoin-project/lotus/files/8859225/miner.log)

Every time in C2 phase，See attachment for details。

### Logging Information

```text
fatal error: unexpected signal during runtime execution
[signal SIGBUS: bus error code=0x2 addr=0x7fe3a6c9beb0 pc=0x7fe3851be6f7]

runtime stack:
runtime.throw({0x3f6887e?, 0x7fe2a4dfcb10?})
	/usr/local/go/src/runtime/panic.go:992 +0x71
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:802 +0x3a9
```

### Repo Steps

1.lotus daemon
2.lotus-miner run

**Comment by ChaiWenyu1983** [comment]
It can be ruled out that it is a memory problem, because I used "memtester" to test all night, and the results showed that the memory is OK.

**Comment by rjan90** [comment]
Hey @ChaiWenyu1983! What kind of hardware specs do you have? (CPU, GPU & RAM). And also how did you build Lotus:
```
'+16-bit-insts' is not a recognized feature for this target (ignoring feature)
'+add-no-carry-insts' is not a recognized feature for this target (ignoring feature)
```
These logs make me suspect that you might not have built Lotus properly for your hardware maybe.

And also, what is the full lotus version - `lotus -v`?

**Comment by ChaiWenyu1983** [comment]
Hey man,
&nbsp;&nbsp;lotus -v:lotus version 1.15.2+mainnet+git.518dc962e







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月10日(星期五) 下午2:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] lotus-miner  Unexpected termination，ask for help。 (Issue #8821)





 
Hey @ChaiWenyu1983! What kind of hardware specs do you have? (CPU, GPU &amp; RAM). And also how did you build Lotus:
 '+16-bit-insts' is not a recognized feature for this target (ignoring feature) '+add-no-carry-insts' is not a recognized feature for this target (ignoring feature)  
These logs make me suspect that you might not have built Lotus properly for your hardware maybe.
 
And also, what is the full lotus version - lotus -v?

**Comment by rjan90** [comment]
> What kind of hardware specs do you have? (CPU, GPU & RAM)


What about this @ChaiWenyu1983? I suspect that it might be a build issue, so building from source again might fix it. Follow this: https://lotus.filecoin.io/storage-providers/operate/upgrades/#upgrade-in-place

**Comment by ChaiWenyu1983** [comment]
CPU:AMD3960X
GPU:TUF-RTX3070
RAM:kingstong 32G*8=256G




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月10日(星期五) 下午5:55
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] lotus-miner  Unexpected termination，ask for help。 (Issue #8821)





  
What kind of hardware specs do you have? (CPU, GPU &amp; RAM)
  
What about this @ChaiWenyu1983? I suspect that it might be a build issue, so building from source again might fix it. Follow this: https://lotus.filecoin.io/storage-providers/operate/upgrades/#upgrade-in-place

**Comment by rjan90** [comment, marked as answer]
Okay, I would consider changing the GPU if you want to do C2 jobs on the GPU, as the Commit 2 job needs more then 10GiB VRAM, so it will definitely fall back to use the CPU for the C2 job as is now. 

So a couple more questions, and some things I would like you to do to troubleshoot more:

Questions:
- Is the specs for the C2-worker only? Or is this machine running everything (lotus, lotus-miner, workers, etc?)
- Are you using CUDA or OpenCL?
- What env-variables have you set?

Recommended things:
- Build from source: https://lotus.filecoin.io/storage-providers/operate/upgrades/#upgrade-in-place
- Change GPU to a GPU with higher VRAM (and not a LHR-card) if you want to do C2-jobs on them.

Also, I´m moving this issue to a discussion, as that is a better suited place for this kind of debugging.
